### PR TITLE
Handle multiple AMIs for appversion (different AMIs in each region)

### PR DESCRIPTION
**Pull request by robfletcher**
I know this should not strictly be necessary but it's pretty easy to make Keel resilient to this case

### CODE REVIEW

File: keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt [hunk 1]
```diff
@@ -59,10 +59,13 @@ class ImageService(
...
-   * Get the latest named image for a package
+   * Get the latest named image for a package.
+   *
+   * @param region if supplied the latest image in this region is returned, if `null` the latest
+   * image regardless of region.
    */
-  suspend fun getLatestNamedImage(packageName: String, account: String): NamedImage? =
-    cloudDriverService.namedImages(DEFAULT_SERVICE_ACCOUNT, "$packageName-", account)
+  suspend fun getLatestNamedImage(packageName: String, account: String, region: String? = null): NamedImage? =
+    cloudDriverService.namedImages(DEFAULT_SERVICE_ACCOUNT, "$packageName-", account, region)
```
Comment (robfletcher) [inline keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt]: CloudDriver already allows filtering by region

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ImageResolver.kt [hunk 2]
```diff
@@ -47,12 +48,12 @@ class ImageResolver(
         }
         val account = dynamicConfigService.getConfig("images.default-account", "test")
         val namedImage = imageService
-          .getLatestNamedImage(artifactName, account) ?: throw NoImageFound(artifactName)
+          .getLatestNamedImage(artifactName, account, region) ?: throw NoImageFound(artifactName)
```
Comment (robfletcher) [inline keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ImageResolver.kt]: supplying the cluster resource's region here means we'll get a filtered list of AMIs back from River

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt [hunk 3]
```diff
@@ -296,7 +296,8 @@ data class ResourceCheckError(
   override val name: String,
   override val application: String,
   override val timestamp: Instant,
-  val exception: Throwable
+  val exceptionType: Class<Throwable>,
+  val exceptionMessage: String?
```
Comment (robfletcher) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt]: Semi-related change. I noticed the resource history has big blobs of stack trace when there are resource check errors. Seems undesirable to store that. I'll need to flush DB before rolling this out.

**Comment by emjburns** [comment]
Is there a case where we deploy a cluster with a different different image in each region?

**Comment by robfletcher** [comment]
> Is there a case where we deploy a cluster with a different different image in each region?

I don't believe so yet. Once we get to the concept of regional roll-outs of new appversions I think that will be a thing.

**Pull request merged**